backend/tests/fixtures/models.py

```diff
@@ -0,0 +1,36 @@
+from collections.abc import AsyncGenerator, Generator
+
+import pytest
+import pytest_asyncio
+from faker import Faker
+
+from src.models import Performance, PerformanceRole, User
+from tests.factories.models import (
+    PerformanceFactory,
+    PerformanceRoleFactory,
+    UserFactory,
+)
+
+
+@pytest.fixture
+def password() -> Generator[str | None] | str:
+    faker = Faker()
+    yield faker.password()
+
+
+@pytest_asyncio.fixture
+async def user(password: str) -> AsyncGenerator[User, None] | User:
+    user = await UserFactory.create_async(password=password)
+    yield user
+
+
+@pytest_asyncio.fixture
+async def performance(user: User) -> AsyncGenerator[Performance, None] | Performance:
+    performance = await PerformanceFactory.create_async(director_id=user.id)
```

Comment:
```suggestion
async def performance(user: User) -> AsyncGenerator[Performance, None]:
    performance = await PerformanceFactory.create_async(director_id=user.id)
```

you're yield-ing value from you function -> it means it can only return the Generator object (AsyncGenerator in your case)